### PR TITLE
Create new content records with null value for published up and down

### PR DESCRIPTION
**Pull request by richard67**
Pull Request for [https://github.com/joomla/joomla-cms/pull/26295](https://github.com/joomla/joomla-cms/pull/26295).

Changes done in the same way as in [https://github.com/joomla/joomla-cms/pull/24675](https://github.com/joomla/joomla-cms/pull/24675) for com_contact and partly taken from Hannes' PR [https://github.com/joomla/joomla-cms/pull/25760](https://github.com/joomla/joomla-cms/pull/25760).

This makes sure new articles are created with null value for all nullable columns and sets the modified time to created time if it is null.

### CODE REVIEW

File: libraries/src/Table/Content.php [hunk 1]
```diff
@@ -252,31 +252,31 @@ public function check()
...
 			$this->publish_up = $this->publish_down;
 			$this->publish_down = $temp;
 		}
 
-		// Set modified to null date if not set
+		// Set modified to created date if not set
 		if (!$this->modified)
 		{
-			$this->modified = $this->_db->getNullDate();
+			$this->modified = $this->created;
```
Comment (wilsonge) [inline libraries/src/Table/Content.php]: This won't work - `$this->created` is only set in the store method :/ We'll have to move this down there (somewhere around line 347) - or probably better move that check into this method - as it is a check

Comment (richard67) [reply libraries/src/Table/Content.php]: > or probably better move that check into this method - as it is a check

Sorry, don't understand which check in which method.

Comment (richard67) [reply libraries/src/Table/Content.php]: You mean moving the complete `if ($this->id)` stuff including the `else` part from method `store()` up to method `check()`? Would that work?

Comment (richard67) [reply libraries/src/Table/Content.php]: First way is easier and I think I've already seen that somewhere else.

**Pull request merged**